MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -1757,6 +1757,20 @@
 MVKDevice::MVKDevice(MVKPhysicalDevice* physicalDevice, const VkDeviceCreateInfo* pCreateInfo) {
 
 	initPerformanceTracking();
+    
+#if MVK_MACOS
+    //on mac OS, the wrong GPU will drive the screen (graphics switching will not occur)
+    //unless we call this specific MTLCreateSystemDefaultDevice method to create the metal device
+    id<MTLDevice> device = physicalDevice->getMTLDevice();
+    if (!device.headless && !device.lowPower) {
+        id<MTLDevice> sysDefaultDevice = MTLCreateSystemDefaultDevice();
+        
+        //lets be 100% sure this is the device the user asked for
+        if (sysDefaultDevice.registryID == device.registryID) {
+            physicalDevice->replaceMTLDevice(sysDefaultDevice);
```

Comment:
Is the requirement for this swap because Metal returns a different `MTLDevice` instance from `MTLCreateSystemDefaultDevice` for the same GPU that it returned in `MTLCopyAllDevices`?

Reply:
Sorry, accidentally resolved.
Yes that’s correct, see the conversation in the attached issue.